modules/thermo-cycler/thermo-cycler-arduino/thermistorsadc.cpp

```diff
@@ -97,12 +97,20 @@ float ThermistorsADC::heat_sink_temperature() {
 
 
 int ThermistorsADC::_read_adc(int index) {
+  int res = 0;
```

Comment:
This, and the return type of `_read_adc`, should be `uint16_t` or some other unsigned type, and the result of `readADC_SingleEnded` should be immediately cast to unsigned - the ADS1115 in single-ended mode only reads voltages above ground reference. We may not have experienced this based on what we're setting the pga to, but if the read voltage was high enough that it had the MSB set we'd get a negative value.

Reply:
I don't think we will get a -ve value. In single ended mode,  the max value we get is 7FFF.. an input beyond this value is clipped to 7FFF.
Also, even though the single ended mode reads voltages above ground ref, the conversion may result in -ve value as per this note in the datasheet:
"Single-ended signal measurements, where VAINN = 0 V and VAINP = 0 V to +FS, only use
the positive code range from 0000h to 7FFFh. However, because of device offset, the
ADS111x can still output negative codes in case VAINP is close to 0 V."

(I can vouch for that note 'cause I just spent an insane amount of time trying to figure out how I was getting occasional -ve values for a pin I had connected to GND 😑)

Reply:
Oof. That’s kinda terrible! But it all makes sense now. I’ll resolve.